include/parsed_tools/non_matching_coupling.h

```diff
@@ -272,17 +275,38 @@ namespace ParsedTools
   void
   NonMatchingCoupling<dim, spacedim>::assemble_matrix(MatrixType &matrix) const
   {
+    const auto &space_mapping    = space_cache->get_mapping();
     const auto &embedded_mapping = embedded_cache->get_mapping();
-    dealii::NonMatching::create_coupling_mass_matrix(*space_cache,
-                                                     *space_dh,
+    if (coupling_type == CouplingType::approximate_L2)
+      {
+        dealii::NonMatching::create_coupling_mass_matrix(*space_cache,
+                                                         *space_dh,
+                                                         *embedded_dh,
+                                                         embedded_quadrature,
+                                                         matrix,
+                                                         *space_constraints,
+                                                         space_mask,
+                                                         embedded_mask,
+                                                         embedded_mapping,
+                                                         *embedded_constraints);
+      }
+    else if (coupling_type == CouplingType::exact_L2)
+      {
+        const double tol            = 1e-9;
+        const auto &cells_and_quads = dealii::NonMatching::compute_intersection(
+          *space_cache, *embedded_cache, this->quadrature_order, tol);
+
+        dealii::NonMatching::assemble_coupling_exact(*space_dh,
```

Comment:
since you are doing this here as well, can you rename also this function? `assemble_coupling_exact` -> `assemble_coupling_mass_matrix_with_exact_intersections`